### PR TITLE
RTK-query

### PR DESCRIPTION
**Pull request by juhoeerola**
Testing if using RTK-query would be feasible refactor. Conclusion: it would be.

I think this is at a good state to consider now. The two remaining items in the todo are relatively independent changes. I'd leave those as separate tasks, and make sure the current implementation is working fine before going further with those.

The user routes require some additional care, as those handle the auth state, and I'm not even quite sure if it is worth moving to use rtk-query.

todo:
- [x] add permission checks back to avoid unnecessary queries
- [x] add null safety back (don't query with `undefined` roadmap id)
- [ ] refactor user routes
- [x] use selectFromQuery to select subset of query (e.g. single roadmap)
- [ ] maybe refactor some local copies to use optimistic updates instead

### CODE REVIEW

File: frontend/src/components/RatingTableComplexity.tsx [hunk 1]
```diff
@@ -23,8 +23,11 @@ const TableComplexityRatingRow: RatingRow = ({
   onEdit,
 }) => {
   const roadmapId = useSelector(chosenRoadmapIdSelector);
-  const { data } = apiV2.useGetRoadmapUsersQuery(roadmapId ?? skipToken);
-  const user = data?.find(({ id }) => id === rating.createdByUser);
+  const { user } = apiV2.useGetRoadmapUsersQuery(roadmapId ?? skipToken, {
+    selectFromResult: ({ data }) => ({
+      user: data?.find(({ id }) => id === rating.createdByUser),
+    }),
```
Comment (AleksiKytola) [inline frontend/src/components/RatingTableComplexity.tsx:29]: Mietin että pitäisikö tällaiset yleisesti käytetyt "selectorit" koota jonnekkin yhteiseen moduliin

**Review by AleksiKytola — COMMENTED** [review]
Looking good so far. I can't really find anything to add other than perhaps separating the often used 'selectors' for `selectFromResult` into a module.

File: frontend/src/components/RatingTableComplexity.tsx (hunk 1 above)
Comment (juhoeerola) [reply frontend/src/components/RatingTableComplexity.tsx]: Lisäsin id:lle kun se oli oikeastaan ainoa yleisemmin käytetty selektori, toi ainoastaan muuttaa `data` kentän ja välittää muut sellaisenaan

**Comment by juhoeerola** [comment]
did rebase & merge locally

**Pull request closed without merging**